source/fundamentals/crud/read-operations/sort.txt

```diff
@@ -4,135 +4,382 @@ Sort Results
-    Document{{_id=3, title=Atlas Shrugged, length=1088, author=Rand}}
-    Document{{_id=4, title=Infinite Jest, length=1104, author=Wallace}}
-    Document{{_id=6, title=A Dance with Dragons, length=1104, author=Martin}}
-    Document{{_id=2, title=Les Misérables, length=1462, author=Hugo}}
+   import static com.mongodb.client.model.Sorts.ascending;
 
-For additional information on the classes and methods mentioned on this
-page, see the following API documentation:
+   // <MongoCollection setup code here>
+   
+   collection.find().sort(ascending("letter", "_id"));
 
-- :java-sync-api:`FindIterable <com/mongodb/client/FindIterable.html>`
-- :java-sync-api:`MongoCursor <com/mongodb/client/MongoCursor.html>`
-- :java-docs:`Sorts </apidocs/mongodb-driver-core/com/mongodb/client/model/Sorts.html>`
+The code snippet above returns the documents in the
+:ref:`sort_example collection <sorts-crud-sort-example>`  
+in the following order: 
+
+.. code-block:: json
+
+   {"_id": 2, "letter": "a", "food": "donuts and coffee"}
+   {"_id": 3, "letter": "a", "food": "maple syrup"}
+   {"_id": 5, "letter": "a", "food": "milk and cookies"}
+   {"_id": 4, "letter": "b", "food": "coffee with sugar"}
+   {"_id": 1, "letter": "c", "food": "coffee with milk"}
+   {"_id": 6, "letter": "c", "food": "maple donut"}
+
+Combining Sort Criteria
+~~~~~~~~~~~~~~~~~~~~~~~
+
+To combine sort criteria, use the ``Sorts.orderBy()`` static factory
+method. The ``orderBy()`` method builds sort criteria that apply passed 
+in sort criteria from left to right in the event of ties. 
+
+In the following code snippet, we use the ``orderBy()`` method to combine a
+descending sort on the ``letter`` field with an ascending sort on the
+``_id`` field.
+
+.. code-block:: java
+
+   import static com.mongodb.client.model.Sorts.orderBy;
+   import static com.mongodb.client.model.Sorts.ascending;
+   import static com.mongodb.client.model.Sorts.descending;
+
+   // <MongoCollection setup code here>
+
+   Bson orderBySort = orderBy(descending("letter"), ascending("_id"));
+   collection.find().sort(orderBySort);
+
+The code snippet above returns the documents in the
+:ref:`sort_example collection <sorts-crud-sort-example>`
+in the following order: 
+
+.. code-block:: json
+
+   {"_id": 1, "letter": "c", "food": "coffee with milk"}
+   {"_id": 6, "letter": "c", "food": "maple donut"}
+   {"_id": 4, "letter": "b", "food": "coffee with sugar"}
+   {"_id": 2, "letter": "a", "food": "donuts and coffee"}
+   {"_id": 3, "letter": "a", "food": "maple syrup"}
+   {"_id": 5, "letter": "a", "food": "milk and cookies"}
+
+.. _sorts-crud-text-search:
+
+Text Search
+-----------
+
+You can specify the order of the results of a 
+:manual:`text search </text-search/>` by how closely they match your
+search string. Each of your search results has a
+:manual:`text score </reference/operator/aggregation/meta/#exp._S_meta>`, a
+numerical value indicating how well that result matches your search. 
```

Comment:
```suggestion
numerical value that indicates how closely that text value matches your search. 
```

Reply:
```suggestion
numerical value that indicates how closely that result matches your search. 
```

Thanks for the note. I agree that "closely" is a better word than "well" here, and I agree that "indicating" should be split into "that indicates", however I disagree that "result" should be changed to "text value". The reason for this is that the text search is by document rather than by field, so I think it is more accurate to that the text score is saying how well a search matched an individual result.

Reply:
> the text search is by document rather than by field
Isn't it by one or more fields as specified by the text index? https://docs.mongodb.com/manual/text-search/#text-index

Reply:
Yes, it is by one or more fields as specified by the text index. My choice of word "document" earlier was imprecise, apologies. I believe the word "result" better conveys the possible multi-field search happening in a `$text` operation better than "text value" which to me implies a single string value in a field. If you disagree with this assessment, I can change "result" to "text value", as both seem appropriate.